ipu-plugin/pkg/ipuplugin/ipuplugin.go

```diff
@@ -75,6 +78,52 @@ func NewIpuPlugin(port int, brCtlr types.BridgeController, p4rtbin string,
 	}
 }
 
+func waitForInfraP4d() (string, error) {
+	ctx := context.Background()
+	// Wait for 300 x 3 secs total
+	maxRetries := 300
+	retryInterval := 3 * time.Second
+
+	var err error
+	var count int
+	var conn *grpc.ClientConn
+	port := "localhost:9559"
+
+	for count = 0; count < maxRetries; count++ {
+		time.Sleep(retryInterval)
+		conn, err = grpc.Dial(port, grpc.WithTransportCredentials(insecure.NewCredentials()))
+		log.Infof("Connecting to server %s retry count:%d", port, count)
+		if err != nil {
+			log.Warnf("Cannot connect to server: %v", err)
+			continue
+		}
+		c := p4_v1.NewP4RuntimeClient(conn)
+		req := &p4_v1.GetForwardingPipelineConfigRequest{
+			DeviceId:     1,
+			ResponseType: p4_v1.GetForwardingPipelineConfigRequest_ResponseType(p4_v1.GetForwardingPipelineConfigRequest_ALL),
+		}
+		fwdResp, err := c.GetForwardingPipelineConfig(ctx, req)
+		if err != nil {
+			log.Warnf("error when retrieving forwardingpipeline config: %v", err)
+			continue
+		}
+
+		config := fwdResp.GetConfig()
+		if config == nil {
+			// pipeline doesn't have a config yet
+			log.Warnf("No forwardingpipeline config yet: %v", err)
+			continue
+		} else {
+			break
+		}
+	}
+	defer conn.Close()
+	if count == maxRetries {
+		return "", fmt.Errorf("Failed to wait for infrap4d")
```

Comment:
Can we print log.Errorf message, before returning.

Reply:
Done